products/azure_rm_batchaccount/api.yaml

```
@@ -0,0 +1,170 @@
+            id_portion: batchAccounts
+            python_variable_name: account_name
+            go_variable_name: accountName
+      read: !ruby/object:Api::Azure::SDKOperationDefinition
+        go_func_name: Get
+        python_func_name: get
+        request:
+          'resourceGroupName': !ruby/object:Api::Azure::SDKTypeDefinition::StringObject
+            id_portion: resourceGroups
+            python_variable_name: resource_group_name
+            go_variable_name: resourceGroup
+          'accountName': !ruby/object:Api::Azure::SDKTypeDefinition::StringObject
+            id_portion: batchAccounts
+            python_variable_name: account_name
+            go_variable_name: accountName
+        response:
+
+    description: |
+      Manage Azure BatchAccount instance.
+    properties:
+      - !ruby/object:Api::Type::ResourceGroup
+        name: 'resourceGroup'
+        description: 'The name of the resource group that contains the Batch account.'
+        required: true
+        azure_sdk_references: ['resourceGroupName']
+      - !ruby/object:Api::Type::String
+        name: 'name'
+        description: 'A name for the Batch account which must be unique within the region. Batch account names must be between 3 and 24 characters in length and must use only numbers and lowercase letters. This name is used as part of the DNS name that is used to access the Batch service in the region in which the account is created. For example: http://accountname.region.batch.azure.com/.'
+        required: true
+        azure_sdk_references: ['accountName']
+      - !ruby/object:Api::Type::String
+        name: 'location'
+        description: 'The region in which to create the account.'
+        required: true
+        azure_sdk_references: ['/location']
+      - !ruby/object:Api::Type::Complex
+        name: 'tags'
+        description: 'The user-specified tags associated with the account.'
+        required: true
+        azure_sdk_references: ['/tags']
+      - !ruby/object:Api::Type::Complex
+        name: 'autoStorage'
+        description: 'The properties related to the auto-storage account.'
+        required: true
+        azure_sdk_references: ['/properties/autoStorage']
+        properties:
+          - !ruby/object:Api::Type::String
+            name: 'storageAccountId'
+            description: 'The resource ID of the storage account to be used for auto-storage account.'
+            required: true
+            azure_sdk_references: ['/properties/autoStorage/storageAccountId']
+      - !ruby/object:Api::Type::Complex
+        name: 'poolAllocationMode'
+        description: 'The pool allocation mode also affects how clients may authenticate to the Batch Service API. If the mode is BatchService, clients may authenticate using access keys or Azure Active Directory. If the mode is UserSubscription, clients must use Azure Active Directory. The default is BatchService.'
+        required: true

```

You can declare enumerations using the following syntax.

```suggestion
        values:
          - :BatchService
          - :UserSubscription
        default_value: :BatchService
```

---

added enumerations